### PR TITLE
BUG: read_json silently ignores the dtype when engine=pyarrow

### LINKED ISSUE #59516: BUG: read_json silently ignores the dtype when engine=pyarrow
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import json

with open("test.json", "w") as f:
    json.dump({"Col1": 123}, f)

dtype = {"Col1": "int32[pyarrow]"}

df = pd.read_json("test.json", dtype=dtype, lines=True, engine="pyarrow", dtype_backend="pyarrow")

print(df.dtypes.to_dict())  # prints {'Col1': int64[pyarrow]}
```

### Issue Description

[The call to `pyarrow_json.read_json`](https://github.com/pandas-dev/pandas/blob/main/pandas/io/json/_json.py#L942) doesn't pass any [`ParseOptions`](https://arrow.apache.org/docs/python/generated/pyarrow.json.ParseOptions.html#pyarrow.json.ParseOptions). This means that determining the dtype is always left up to the pyarrow json parser's dtype inference system, even when we explicitly passed a requested dtype into `pd.read_json`.

### Expected Behavior

`read_json`'s pyarrow parser should respect the requested `dtype`.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.12.3.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 11
Version               : 10.0.22631
machine               : AMD64
processor             : Intel64 Family 6 Model 165 Stepping 5, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : None
LOCALE                : English_United Kingdom.1252

pandas                : 2.2.2
numpy                 : 2.0.1
pytz                  : 2024.1
dateutil              : 2.9.0.post0
setuptools            : 72.1.0
pip                   : None
Cython                : None
pytest                : 8.3.2
hypothesis            : None
sphinx                : 7.4.7
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : 3.1.4
IPython               : 8.26.0
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : 2024.6.1
gcsfs                 : None
matplotlib            : 3.9.1
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : 17.0.0
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.14.0
sqlalchemy            : None
tables                : None
tabulate              : 0.9.0
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by jahn96**
- [x] closes #59516  (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -2286,3 +2287,25 @@ def test_read_json_lines_rangeindex():
     result = read_json(StringIO(data), lines=True).index
     expected = RangeIndex(2)
     tm.assert_index_equal(result, expected, exact=True)
+
+
```
Comment (yuanx749) [inline pandas/tests/io/json/test_pandas.py:2291]:
```suggestion

@td.skip_if_no("pyarrow")
```

**Review by yuanx749 — CHANGES_REQUESTED** [review]

File: pandas/tests/io/json/test_pandas.py [hunk 2]
```diff
@@ -2286,3 +2287,25 @@ def test_read_json_lines_rangeindex():
...
+def test_read_json_pyarrow_dtype(datapath):
```
Comment (yuanx749) [inline pandas/tests/io/json/test_pandas.py:2292]:
```suggestion
@td.skip_if_no("pyarrow")
def test_read_json_pyarrow_dtype(datapath):
```

**Review by yuanx749 — CHANGES_REQUESTED** [review]

File: pandas/tests/io/json/test_pandas.py (hunk 2 above)
Comment (jahn96) [reply pandas/tests/io/json/test_pandas.py:2293]: Addressed [here](https://github.com/pandas-dev/pandas/pull/59815/commits/f78c8115b29b2213b48f220777db31c80c2a2ec0)

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**